Source/WebKit/UIProcess/API/Swift/WebPage.swift

```diff
@@ -120,20 +125,125 @@ public class WebPage_v0 {
             createObservation(for: \.isLoading, backedBy: \.isLoading),
             createObservation(for: \.serverTrust, backedBy: \.serverTrust),
             createObservation(for: \.hasOnlySecureContent, backedBy: \.hasOnlySecureContent),
+            createObservation(for: \.isWritingToolsActive, backedBy: \.isWritingToolsActive),
             createObservation(for: \.themeColor, backedBy: \.themeColor),
         ]
     }
 
+    public static func handlesURLScheme(_ scheme: String) -> Bool {
+        WKWebView.handlesURLScheme(scheme)
+    }
+
     @discardableResult
     public func load(_ request: URLRequest) -> NavigationID? {
         backingWebView.load(request).map(NavigationID.init(_:))
     }
 
+    @discardableResult
+    public func load(_ data: Data, mimeType: String, characterEncoding: String.Encoding, baseURL: URL) -> NavigationID? {
+        let cfEncoding = CFStringConvertNSStringEncodingToEncoding(characterEncoding.rawValue)
+        guard cfEncoding != kCFStringEncodingInvalidId else {
+            preconditionFailure("\(characterEncoding) is not a valid character encoding")
```

Comment:
Are the supported encodings fixed by the OS? Making this a precondition means that we’re calling it a programmer error, but I’m wondering how easy it is for developers to get right.

Put another way, would we throw an `NSException` for this in ObjC?

Reply:
In actuality, I don't think this case would ever happen since devs would 99% use one of the static properties of String.Encoding, so I just put that precondition there just in case.

cc @beidson for the actual answers to the questions though